Src/Support/Google.Apis/Services/BaseClientService.cs

```diff
@@ -95,10 +95,22 @@ public class Initializer
             /// <summary>Gets or sets the API Key. Default value is <c>null</c>.</summary>
             public string ApiKey { get; set; }
 
+            private string _applicationName;
             /// <summary>
             /// Gets or sets Application name to be used in the User-Agent header. Default value is <c>null</c>. 
             /// </summary>
-            public string ApplicationName { get; set; }
+            public string ApplicationName
+            {
+                get => _applicationName;
+                set
+                {
+                    if (value != null && !ProductInfoHeaderValue.TryParse(value, out var _))
```

Comment:
It's `System.Net.Http.Headers.ProductInfoHeaderValue`

Thanks for pointing out the back-compat problem. Validation moved to in the `BaseClientService` constructor, after which `ApplicationName` cannot be changed.